Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverInformation.hs

```diff
@@ -101,6 +102,34 @@ updateEnabledVerifiedState driverId isEnabled isVerified = do
         <> [DriverInformationLastEnabledOn =. val (Just now) | isEnabled]
     where_ $ tbl ^. DriverInformationDriverId ==. val (toKey $ cast driverId)
 
+updateDynamicBlockedState :: Id Person.Driver -> Maybe Text -> Maybe Int -> Bool -> SqlDB ()
+updateDynamicBlockedState driverId blockedReason Nothing isBlocked = do
+  now <- getCurrentTime
+  Esq.update $ \tbl -> do
+    set
+      tbl
+      ( [ DriverInformationBlocked =. val isBlocked,
+          DriverInformationBlockedReason =. val blockedReason,
+          DriverInformationUpdatedAt =. val now
+        ]
+          <> [DriverInformationNumOfLocks +=. val 1 | isBlocked]
+      )
+    where_ $ tbl ^. DriverInformationDriverId ==. val (toKey $ cast driverId)
+updateDynamicBlockedState driverId blockedReason blockedExpiryTime isBlocked = do
```

Comment:
NOTE: this syntax is actually a way of case match and once you have handled `Nothing` case above, you could have written `Just blockedExpiryTime` instead of `blockedExpiryTime` then you wouldn't have to handle maybe value again in the fn body -> _addUTCTime (maybe 0 (\x -> fromIntegral x * 60 * 60) blockedExpiryTime) now_

Reply:
ok